text/0000-best-practices-and-guidelines.md

```diff
@@ -0,0 +1,100 @@
+# Meta
+[meta]: #meta
+- Name: Best practices and guidelines for Cloud Native Buildpacks
+- Start Date: 2021-03-29
+- Author(s): [@samj1912](https://github.com/samj1912)
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+[summary]: #summary
+
+As the CNB (Cloud Native Buildpacks) community grows and more people and organizations adopt it, there is a need for a set of guidelines that augment the CNB `spec` with a set of recommendations on how to design and implement -
+
+* Buildpacks
+* Platforms
+
+These guidelines should not in any way be considered a mandate that `buildpack` and `platform` authors and users must follow, though they may find that those that conform well to these guidelines integrate better with the existing CNB ecosystem than those that do not.
+
+
+# Motivation
+[motivation]: #motivation
+
+> **NOTE** : The examples below are theoretical guidelines that are meant to serve as examples and are not meant to be the actual guidelines.
+
+Some of the motivations for the creating a set of guidelines are -
+
+## Better consistency and user experience for users and authors
+
+Examples - 
+
+- A consistent naming scheme for `buildpacks`, `stack` IDs
+- A consistent way to "configure" `buildpacks` like specifying environment variables through a project descriptor.
+
+## Transfer of knowledge and best pratices accrued over time by experienced users to newer users
+
+Examples - 
+
+- Best practices around layer cache invalidation
+- Best practices for a `platform` operator
+
+
+## Better portability and reusability amongst different CNB ecosystems
+
+Examples - 
+
+- Using `platform` agnostic constructs when writing and distributing `buildpacks`
+
+
+## Allowing users to write maintainable and efficient `buildpacks` and `platforms`
+
+Examples - 
+
+- Creation of buildpacks that attempt to do one thing and only one thing well and composing them using `meta-buildpacks` (Paketo ideology)
+
+## Avoiding common pitfalls that new users may be unaware of
+
+Examples - 
+
+- `Bill-of-materials` needs to be regenerated on each run
+
+
+# What it is
+[what-it-is]: #what-it-is
+
+The `guidelines` would be contained in a new repository called `guidelines` which will be created in the `buildpacks` Github organization. This repository will be maintained by the `Learning team`. The repository should have seperate sections that target different personas and list guidelines and best practices specific to them -
```

Comment:
Would you prefer a separate repo just putting it on `buildpacks/docs` as a section?

Reply:
Yup - added that as an alternative. From the last learning team sync - others seemed in favor of that as well.